library/Freckle/App/Exception/MonadThrow.hs

```diff
@@ -0,0 +1,91 @@
+module Freckle.App.Exception.MonadThrow
+  ( throwM
+  , throwString
+  , fromJustNoteM
+  , impossible
+  , catch
+  , catchJust
+  , catches
+  , try
+  , checkpointCallStack
+
+    -- * Miscellany
+  , MonadThrow
+  , MonadCatch
+  , MonadMask
+  , module Freckle.App.Exception.Types
+  ) where
+
+import Freckle.App.Exception.Types
+
+import Control.Applicative (pure)
+import Control.Monad.Catch (MonadCatch, MonadMask, MonadThrow)
+import Data.Either (Either (..))
+import Data.Function (($), (.))
+import Data.Functor (fmap)
+import Data.Maybe (Maybe, fromMaybe, maybe)
+import Data.String (String)
+import GHC.IO.Exception (userError)
+import GHC.Stack (withFrozenCallStack)
+
+import qualified Control.Exception.Annotated as Annotated
+import qualified Control.Monad.Catch
+
+-- Throws an exception, wrapped in 'AnnotatedException' which includes a call stack
+throwM
+  :: forall e m a. HasCallStack => MonadThrow m => Exception e => e -> m a
+throwM = Annotated.throw
+
+throwString :: forall m a. HasCallStack => MonadThrow m => String -> m a
+throwString = throwM . userError
+
+fromJustNoteM :: (HasCallStack, MonadThrow m) => String -> Maybe a -> m a
+fromJustNoteM err = maybe (throwString err) pure
+
+impossible :: (HasCallStack, MonadThrow m) => m a
+impossible = throwString "Impossible"
+
+catch :: (HasCallStack, Exception e, MonadCatch m) => m a -> (e -> m a) -> m a
+catch = Annotated.catch
```

Comment:
I'd recommend using `withFrozenCallStack` here to avoid some callstack noise

```suggestion
catch :: (HasCallStack, Exception e, MonadCatch m) => m a -> (e -> m a) -> m a
catch = withFrozenCallStack Annotated.catch
```

Reply:
Thanks!